### ISSUE TITLE
Training/Validation loss  become  "nan"

### ISSUE THREAD

**Issue by ccyang123** [wponghiran/imp-snns-for-sl#1]
@wponghiran

Thank you for providing your great work to github. 
I am learning how to use SNN to do the ASR work. 
I got a problem when executing the  improved_spiking_model_v2 model with LibriSpeech. 

After I run the following command, I found that the Training/Validation loss becomes "nan" from Epoch 04/23. 
(The final WER is 15.72%, which is not consistent with the paper (18.72%) )

Can you give me some advice to solve this problem?  Thank you!

(1) The command I used: 
linux%> python  run_exp.py cfg/LibriSpeech_improved_spiking_model_v2_l2_e24_nb6th3_adam1.cfg

(2) The training log is listed below.

------------------------------ Epoch 03 / 23 ------------------------------
Training train_clean_100 chunk = 1 / 1
[========================================] 100% Training | (Batch 5759/5759)9) | L:2.055
Validating dev_clean chunk = 1 / 1
[=======================================-] 100% Validating | (Batch 47/47))
----- Summary epoch 03 / 23
Training on ['train_clean_100']
Loss = 2.055 | err = 0.523
-----
Validating on dev_clean
Loss = 1.872 | err = 0.490
-----
Learning rate on architecture1 = 0.001
Threshold on architecture1 = 3
-----
Elapsed time (s) = 1274

------------------------------ Epoch 04 / 23 ------------------------------
Training train_clean_100 chunk = 1 / 1
[========================================] 100% Training | (Batch 5759/5759)9) | L:nan
Validating dev_clean chunk = 1 / 1
[=======================================-] 100% Validating | (Batch 47/47))
----- Summary epoch 04 / 23
Training on ['train_clean_100']
Loss = nan | err = 0.926
-----
Validating on dev_clean
Loss = nan | err = 0.912
-----
Learning rate on architecture1 = 0.001
Threshold on architecture1 = 3
-----
Elapsed time (s) = 1271

**Comment by ccyang123** [comment]
After I ran the training job with 1 GPU (instead of 2 GPU), the loss value becomes normal (no nan occurs).  
FYI
